hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/TransitRegionStateProcedure.java

```diff
@@ -200,14 +200,21 @@ private void queueAssign(MasterProcedureEnv env, RegionStateNode regionNode)
     }
   }
 
-  private void openRegion(MasterProcedureEnv env, RegionStateNode regionNode) throws IOException {
+  private void openRegion(MasterProcedureEnv env, RegionStateNode regionNode)
+    throws IOException, ProcedureSuspendedException {
     ServerName loc = regionNode.getRegionLocation();
     if (loc == null) {
       LOG.warn("No location specified for {}, jump back to state {} to get one", getRegion(),
         RegionStateTransitionState.REGION_STATE_TRANSITION_GET_ASSIGN_CANDIDATE);
       setNextState(RegionStateTransitionState.REGION_STATE_TRANSITION_GET_ASSIGN_CANDIDATE);
       return;
     }
+    final boolean isMeta = regionNode.getRegionInfo().isMetaRegion();
+    final boolean isMetaAvailable = !env.getAssignmentManager().isMetaRegionInTransition();
+    if (!isMeta && !isMetaAvailable) {
+      // meta is not assigned yet, so yield
+      throw new ProcedureSuspendedException();
```

Comment:
Good to add a message that open region for ${regionNameAsString} is suspended?

Reply:
This can not solve all the proble, neither the check here, nor the check in waitInitialized. It could always happen that when checking the meta is online,but when you actually write to it, it goes offline... So in general, we should have a way to deal with meta update failure(maybe just a retry at procedure level?)and have a smaller timeout on updaing meta operation.

Reply:
I agree, neither this code nor the proposed change provide a satisfactory solution. However, I think if we can avoid turning the SRSP over to the worker thread when we know meta is unavailable, we reduce the window of risk. Another change be to alter the configuration of the client that attempts the meta update: instead of retrying for the full loop, it might try a single RPC and if that fails yield the procedure. That way the PEWorker isn't occupied by the default client retry logic.